### PR TITLE
ci: use DWARF-4 for Valgrind jobs

### PR DESCRIPTION
**Pull request by fanquake**
[clang-14 defaults to using DWARF-5](https://releases.llvm.org/14.0.0/tools/clang/docs/ReleaseNotes.html#dwarf-support-in-clang), which breaks vlagrinds (3.18) ability
to parse debug info. Valgrind [claims to support DWARF-5](https://valgrind.org/docs/manual/dist.news.html) from version
3.18 onwards, but maybe that only works when compiling with GCC.

Explicitly use DWARF-4 for now. Note that from 11.0 [GCC also defaults to
using DWARF-5](https://www.gnu.org/software/gcc/gcc-11/changes.html).

Also remove a Boost related suppression.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
No objection, but it would be good for either clang or valgrind to fix this, so that we can revert the workaround and that devs are able to use clang-14 out of the box.

**Comment by fanquake** [comment]
Looks like there is at least discussion on the Valgrind mailing list [[Valgrind-users] does Valgrind-3.19.0.GIT support Clang14 dwarf 5?](https://sourceforge.net/p/valgrind/mailman/valgrind-users/thread/AM0PR08MB513876B66ECBBFF37E3AD2FF89329%40AM0PR08MB5138.eurprd08.prod.outlook.com/#msg376098280). I haven't found anything on the LLVM side yet.

**Comment by laanwj** [comment]
I don't know how big of a change it is, but if it's anything like going from DWARF 2 to 3 it's not going to be a trivial change in valgrind. I suspect it's going to take a while to support it.

Could add a comment to mention why this override is needed and what would be the condition to remove it.

Otherwise code review ACK 5691a8f0ec8d0bfc470990dfaed9d230346a116e

File: ci/test/00_setup_env_native_fuzz_with_valgrind.sh [hunk 1]
```diff
@@ -15,5 +15,5 @@ export RUN_FUNCTIONAL_TESTS=false
 export RUN_FUZZ_TESTS=true
 export FUZZ_TESTS_CONFIG="--valgrind"
 export GOAL="install"
-export BITCOIN_CONFIG="--enable-fuzz --with-sanitizers=fuzzer CC=clang CXX=clang++"
+export BITCOIN_CONFIG="--enable-fuzz --with-sanitizers=fuzzer CC=clang CXX=clang++ CXXFLAGS='-fdebug-default-version=4'"
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_fuzz_with_valgrind.sh:18]: Could leave CXXFLAGS untouched to minimize the effective changes?

```suggestion
# Temporarily pin dwarf 4, until valgrind can understand clang's dwarf 5
export BITCOIN_CONFIG="--enable-fuzz --with-sanitizers=fuzzer CC=clang CXX='clang++ -fdebug-default-version=4'"
```

**Comment by fanquake** [comment]
> Could add a comment to mention why this override is needed and what would be the condition to remove it.

Added. 

> Could leave CXXFLAGS untouched to minimize the effective changes?

Done. Used `-gdwarf-4` to minimize even further.

**Comment by fanquake** [comment]
This actually only seems to work when using `CXXFLAGS` and the full `-fdebug-default-version=4`, so I've reverted back to the original change + the comment.

File: ci/test/00_setup_env_native_valgrind.sh [hunk 2]
```diff
@@ -13,4 +13,5 @@ export USE_VALGRIND=1
 export NO_DEPENDS=1
 export TEST_RUNNER_EXTRA="--nosandbox --exclude feature_init,rpc_bind,feature_bind_extra"  # Excluded for now, see https://github.com/bitcoin/bitcoin/issues/17765#issuecomment-602068547
 export GOAL="install"
-export BITCOIN_CONFIG="--enable-zmq --with-incompatible-bdb --with-gui=no CC=clang CXX=clang++"  # TODO enable GUI
+# Temporarily pin dwarf 4, until valgrind can understand clang's dwarf 5
+export BITCOIN_CONFIG="--enable-zmq --with-incompatible-bdb --with-gui=no CC=clang CXX=clang++ CXXFLAGS='-fdebug-default-version=4"  # TODO enable GUI
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_valgrind.sh:17]:
```suggestion
export BITCOIN_CONFIG="--enable-zmq --with-incompatible-bdb --with-gui=no CC=clang CXX=clang++ CXXFLAGS='-fdebug-default-version=4'"  # TODO enable GUI
```

Comment (fanquake) [reply ci/test/00_setup_env_native_valgrind.sh]: 😰

**Review by maflcko — APPROVED** [review]
review ACK

**Pull request merged**